### PR TITLE
Add model metadata

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
Add API for adding a metadata dictionary to Keras\Pytorch\ONNX models

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: mct_quantizers/keras/load_model.py [hunk 1]
```diff
@@ -59,7 +60,18 @@ def keras_load_quantized_model(filepath: str, custom_objects: Any = None, compil
         kwargs = {}
         if options is not None:
             kwargs['options'] = options
-        return tf.keras.models.load_model(filepath, custom_objects=qi_custom_objects, compile=compile, **kwargs)
+
+        # Load model
+        loaded_model = tf.keras.models.load_model(filepath, custom_objects=qi_custom_objects, compile=compile, **kwargs)
+
+        # Extract metadata is exists
```
Comment (ofirgo) [inline mct_quantizers/keras/load_model.py:67]: typo (is --> if)

File: mct_quantizers/keras/load_model.py [hunk 2]
```diff
@@ -59,7 +60,18 @@ def keras_load_quantized_model(filepath: str, custom_objects: Any = None, compil
...
+        metadata_layers = [l for l in loaded_model.layers if isinstance(l, MetadataLayer)]
+        if len(metadata_layers) > 0:
+            if len(metadata_layers) > 1:
+                Logger.warning('Found more than 1 MetadataLayer layers in model.')
```
Comment (ofirgo) [inline mct_quantizers/keras/load_model.py:71]: add "Loading the metadata from the first layer only" or something like that

File: mct_quantizers/keras/metadata.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                Adding author name and model version to model's metadata
+
+                >>> model_with_metadata = add_metadata(model, {'author': 'John Doe', 'model version': 3})
+        """
+        metadata = verify_and_init_metadata(metadata)
+        if FRAMEWORK_VERSION not in metadata:
+            metadata[FRAMEWORK_VERSION] = tf.__version__
+
+        model.metadata_layer = MetadataLayer(metadata=metadata)
+        model.metadata = model.metadata_layer.metadata
```
Comment (ofirgo) [inline mct_quantizers/keras/metadata.py:58]: is this necessary? why the redundancy?
how does that even work? how does the model have "metadata" attribute?

**Review by ofirgo — APPROVED** [review]
Some minor comments.
Additionally - do we want to figure out how to add this to the backward/forward compatibility tests?

File: mct_quantizers/common/metadata.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        Logger.error('metadata dictionary should only have string keys.')
+    if not all([isinstance(v, (str, int, float)) for v in metadata.values()]):
+        Logger.error('metadata dictionary should only have string, integer or float values.')
+
+    if metadata is None:
+        metadata = {}
+    if PYTHON_VERSION not in metadata:
+        metadata[PYTHON_VERSION] = sys.version
+    if MCTQ_VERSION not in metadata:
+        metadata[MCTQ_VERSION] = mct_quantizers.__version__
```
Comment (Chizkiyahu) [inline mct_quantizers/common/metadata.py:46]: in the example model that I got the value was '__version__'
please check that this kine works

File: tests/keras_tests/test_keras_load_model.py [hunk 5]
```diff
@@ -212,3 +213,15 @@ def test_save_and_load_weights_lut_pot(self):
         layer_with_quantizer = KerasQuantizationWrapper(Conv2D(3, 3),
                                                         {'kernel': quantizer})
         self._one_layer_model_save_and_load(layer_with_quantizer)
+
+    def test_save_and_load_metadata(self):
+        _input = tf.keras.layers.Input((8,))
+        model = tf.keras.Model(inputs=_input, outputs = _input + 5)
+        model = add_metadata(model, {'test': 'test123'})
+
+        _, tmp_h5_file = tempfile.mkstemp('.h5')
```
Comment (Chizkiyahu) [inline tests/keras_tests/test_keras_load_model.py:222]: please add also .keras test

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

**Review by reuvenperetz — APPROVED** [review]
